# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/pcre2/hotfix-x32-no-jit.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- pcre2-10.47.orig/deps/sljit/sljit_src/sljitConfigCPU.h	2025-10-20 13:35:04.000000000 +0200
+++ pcre2-10.47/deps/sljit/sljit_src/sljitConfigCPU.h	2026-07-26 15:08:47.516719699 +0200
@@ -105,7 +105,9 @@
 
 #if defined(__i386__) || defined(__i386)
 #define SLJIT_CONFIG_X86_32 1
-#elif defined(__x86_64__)
+#elif defined(__x86_64__) && !defined(__ILP32__)
+/* x32 defines __x86_64__ but has 32 bit long/pointers, so sljit_sw ends up
+   4 bytes wide while the x86-64 backend emits 8 byte immediates. */
 #define SLJIT_CONFIG_X86_64 1
 #elif defined(__aarch64__)
 #define SLJIT_CONFIG_ARM_64 1
